prow/plugins/lgtm/lgtm.go

```diff
@@ -204,3 +207,56 @@ func handle(gc githubClient, log *logrus.Entry, e *event) error {
 	}
 	return nil
 }
+
+type ghLabelClient interface {
+	GetIssueLabels(owner, repo string, number int) ([]github.Label, error)
+	RemoveLabel(owner, repo string, number int, label string) error
+}
+
+func handlePullRequest(gc ghLabelClient, pe github.PullRequestEvent) error {
+	if pe.PullRequest.Merged {
+		return nil
+	}
+
+	if pe.Action != "synchronize" {
+		return nil
+	}
+
+	var (
+		org    = pe.PullRequest.Base.Repo.Owner.Login
+		repo   = pe.PullRequest.Base.Repo.Name
+		number = pe.PullRequest.Number
+	)
+
+	// Check if it has the label before removing to save the API token
+	// that comes with attempting to remove a label.
+	labels, err := gc.GetIssueLabels(org, repo, number)
```

Comment:
Argh, I misunderstood something. I was thinking you could say `if pe.PullRequest.HasLabel("lgtm") { ... }` but the labels are not sent in the `PullRequestEvent`, so no matter what, you have to spend the token. You can undo this change, you were right before that just trying to remove and ignoring the LabelNotFound error is correct.